pandasai/helpers/anonymizer.py

```diff
@@ -0,0 +1,130 @@
+import re
+import random
+import string
+import pandas as pd
+
+
+def is_valid_email(email):
+    """
+    Check if the given email is valid based on regex pattern.
+
+    :param email: str, email address to be checked
+    :return: bool, True if the email is valid, otherwise False
+    """
+    email_regex = r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
+    return re.match(email_regex, email)
+
+
+def is_valid_phone_number(phone_number):
+    """
+    Check if the given phone number is valid based on regex pattern.
+
+    :param phone_number: str, phone number to be checked
+    :return: bool, True if the phone number is valid, otherwise False
+    """
+    pattern = r"\b(?:\+?\d{1,3}[- ]?)?\(?\d{3}\)?[- ]?\d{3}[- ]?\d{4}\b"
+    return re.search(pattern, phone_number)
+
+
+def is_valid_credit_card(credit_card_number):
+    """
+    Check if the given credit card number is valid based on regex pattern.
+
+    :param credit_card_number: str, credit card number to be checked
+    :return: bool, True if the credit card number is valid, otherwise False
+    """
+    pattern = r"^\d{4}[- ]?\d{4}[- ]?\d{4}[- ]?\d{4}$"
+    return re.search(pattern, credit_card_number)
+
+
+def generate_random_email():
+    """
+    Generate a random email address using predefined domains.
+
+    :return: str, generated random email address
+    """
+    domains = [
+        "gmail.com",
+        "yahoo.com",
+        "hotmail.com",
+        "outlook.com",
+        "icloud.com",
+        "aol.com",
+        "protonmail.com",
+        "zoho.com",
+    ]
+    name_length = random.randint(6, 12)
+    domain = random.choice(domains)
+    letters = string.ascii_lowercase + string.digits + "-_"
+    username = "".join(random.choice(letters) for i in range(name_length))
+    email = username + "@" + domain
+    return email
+
+
+def generate_random_phone_number():
+    """
```

Comment:
Noticed here we have a problem: sometimes the script detects a "fake number" (check out the example of the GDP). The result is we convert it the gpd to a phone number like this: `+60 4702356135`, which can be misleading and cause hallucinations to the LLM.

What if we add the country prefix (i.e. +60) only if it was originally present, otherwise we just generate a random number without prefix?

So:
- if the field is, for example, 2411255037952
   - we generate a random number, say 5263101311
- if the field is, for example, +39 11255037952
   - we generate a random number, say +39 5263101311

Reply:
Addressed, thanks. Great catch :D